src/Files.App/Helpers/NaturalStringComparer.cs

```diff
@@ -7,32 +7,110 @@ public sealed class NaturalStringComparer
 	{
 		public static IComparer<object> GetForProcessor()
 		{
-			return Win32Helper.IsRunningOnArm ? new StringComparerArm64() : new StringComparerDefault();
+			return new NaturalComparer(StringComparison.CurrentCulture);
 		}
-
-		private sealed class StringComparerArm64 : IComparer<object>
+		
+		private sealed class NaturalComparer : IComparer<object?>, IComparer<string?>, IComparer<ReadOnlyMemory<char>>
```

Comment:
Can you add xml comment on this class so others can understand what this is doing under the hood.